### DISCUSSION TITLE
Allow author to comment on lines in Pull Request before it is opened

### DISCUSSION THREAD

**Discussion by pg8wood** [community/community#34357] [Pull Requests]
Many times when opening a PR I find myself wanting to add a comment to a line to explain something. I normally open a PR and immediately go to comment on it before anyone starts reviewing it. But, sometimes I'm blessed with swift reviews and someone asks about a line before I'm able to add my comment. It would be nice if the author could add these comments before the PR is opened and able to be reviewed by others.

**Comment by tejohnso** [comment]
Also the intuitive behaviour around comments within a new PR is misleading.

If while you're crafting the PR, you click on a commit below the initial PR comment and add comments that way, those comments will not be visible as part of the PR timeline. I guess they're considered to be commit comments outside the context of a PR in that case, even though they were created while creating a PR. When you submit your PR you won't see them in the PR timeline. You won't even see them if you click on the commit from within the PR timeline. You'll have to click on the small comment bubble icon beside the PR in order to see them.

In short, I attempted a workaround, and it didn't work as expected.

Draft PRs aren't guaranteed to help either because people still end up looking at it before the comments are in and they aren't aware that comments are still coming. I suppose it should be a convention that we don't look at draft PRs unless explicitly asked to. Ideally people wouldn't even be notified of draft PRs but I don't think that's the case. They're going to be notified. Which is confusing. You're notifying me, but you don't want me to look at it yet? Why did you notify me then?
